File: backend/FinanceApp.Domain/Entities/Receivable.cs
using FinanceApp.Domain.Enums;
using FinanceApp.Domain.Interfaces;

namespace FinanceApp.Domain.Entities;

public class Receivable : BaseEntity, IConcurrencyVersioned
{
    /// <summary>乐观并发版本号（由 AppDbContext 自动维护）</summary>
    public long Version { get; set; }

    public long ProjectId { get; set; }
    public long? CustomerId { get; set; }
    public long? SupplierId { get; set; }
    public long? PersonId { get; set; }
    public long? ReceivableTypeId { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal ReceivedAmount { get; set; }
    public decimal RemainingAmount { get; set; }
    public DateTime? DueDate { get; set; }
    public ReceivableStatus Status { get; set; } = ReceivableStatus.Pending;
    public string? Description { get; set; }
    public DateTime? SettledAt { get; set; }

    // Navigation properties
    public Project Project { get; set; } = null!;
    public Customer? Customer { get; set; }
    public Supplier? Supplier { get; set; }
    public Person? Person { get; set; }
    public ReceivableType? ReceivableType { get; set; }
    public ICollection<ReceivableDetail> Details { get; set; } = new List<ReceivableDetail>();
    public ICollection<TagBinding> TagBindings { get; set; } = new List<TagBinding>();
}

File: frontend/src/features/finance/types/receivable.ts
import type { TagItem } from '@/features/master-data/projects/types/project'

export interface Receivable {
  id: number
  projectId: number
  projectName?: string
  customerId?: number
  customerName?: string
  supplierId?: number
  supplierName?: string
  personId?: number
  personName?: string
  receivableTypeId?: number
  receivableTypeName?: string
  totalAmount: number
  receivedAmount: number
  remainingAmount: number
  dueDate?: string
  status: 'pending' | 'partial' | 'settled'
  description?: string
  tags: TagItem[]
  createdAt: string
  updatedAt: string
  settledAt?: string
  details: ReceivableDetail[]
}

export interface ReceivableDetail {
  id: number
  receivableId: number
  transactionId: number
  paymentDate: string
  amount: number
  paymentMethod?: string
  description?: string
  createdAt: string
}

export interface ReceivableSummary {
  totalReceivable: number
  totalReceived: number
  totalRemaining: number
  pendingCount: number
  partialCount: number
  settledCount: number
  overdueCount: number
}

export interface ReceivableTrend {
  months: string[]
  amounts: number[]
}

export interface ReceivableAging {
  categories: string[]
  amounts: number[]
}

export interface CreateReceivableRequest {
  projectId: number
  customerId?: number
  supplierId?: number
  personId?: number
  receivableTypeId?: number
  totalAmount: number
  dueDate?: string
  description?: string
}

export interface UpdateReceivableRequest {
  projectId: number
  customerId?: number
  supplierId?: number
  personId?: number
  receivableTypeId?: number
  totalAmount: number
  dueDate?: string
  description?: string
}

export interface ReceivePaymentRequest {
  paymentDate: string
  amount: number
  paymentMethod?: string
  description?: string
  transactionId: number
}

export interface ReceivableStatistics {
  totalCount: number
  pendingCount: number
  partialCount: number
  settledCount: number
  totalAmount: number
  receivedAmount: number
  remainingAmount: number
  overdueAmount: number
}

export interface ReceivableType {
  id: number
  name: string
  code?: string
  description?: string
  isActive: boolean
  sortOrder: number
}

export interface CreateReceivableTypeRequest {
  name: string
  code?: string
  description?: string
  isActive?: boolean
  sortOrder?: number
}

export interface UpdateReceivableTypeRequest {
  name: string
  code?: string
  description?: string
  isActive?: boolean
  sortOrder?: number
}
